go/samples/coffee-shop/main.go

```diff
@@ -159,12 +146,10 @@ func main() {
 	})
 
 	simpleStructuredGreetingPrompt, err := dotprompt.Define("simpleStructuredGreeting", simpleStructuredGreetingPromptTemplate,
-		dotprompt.Config{
-			Model:        g,
-			InputSchema:  r.Reflect(simpleGreetingInput{}),
-			OutputFormat: ai.OutputFormatJSON,
-			OutputSchema: r.Reflect(simpleGreetingOutput{}),
-		},
+		dotprompt.WithDefaultModel(g),
+		dotprompt.WithInputType(simpleGreetingInput{}),
+		dotprompt.WithOutputFormat(ai.OutputFormatJSON),
```

Comment:
Let's omit the setting of output format here since we set type. We want to show the simplest most canonical ways of doing things.